src/realm/sync/subscriptions.cpp

```diff
@@ -288,7 +283,7 @@ std::pair<SubscriptionSet::iterator, bool> MutableSubscriptionSet::insert_or_ass
     auto table_name = Group::table_name_to_class_name(query.get_table()->get_name());
     auto query_str = query.get_description();
     auto it = std::find_if(begin(), end(), [&](const Subscription& sub) {
-        return (sub.name().empty() && sub.object_class_name() == table_name && sub.query_string() == query_str);
```

Comment:
Was this a bug? Or do we want to treat empty the same as null/have emptyness be how we express optionality?

Reply:
I think this was a bug actually.